src/main/java/net/snowflake/client/jdbc/SnowflakeSQLException.java

```diff
@@ -57,6 +57,13 @@ public SnowflakeSQLException(String reason, String sqlState) {
     logger.debug("Snowflake exception: {}, sqlState:{}", reason, sqlState);
   }
 
+  public SnowflakeSQLException(String queryId, String reason, String sqlState) {
```

Comment:
after adding new constructor - can we mark some existing constructors as deprecated when are not used?